# inladen

In [4]:
import pandas as pd
import numpy as np
from sklearn.metrics import mean_squared_error

In [6]:
df = pd.read_csv('../Data/CSV/cleaned.csv', sep=';', low_memory=False)
print('aantal rijen en kolommen:', df.shape)

aantal rijen en kolommen: (259140, 11)


# Baseline model

Aan de hand van deze columns zullen we de stm_fh_duur nemen als targetvariable en daarvan een baseline model maken.

In [7]:
#
time_spans = df['hersteltijd']
time_spans = time_spans[time_spans > 0].sort_values()
time_spans.value_counts()

3      14474
1      13922
15      6844
30      6726
5       6468
       ...  
356        4
337        4
347        4
344        3
358        1
Name: hersteltijd, Length: 358, dtype: int64

We zien dat 3 minuten het vaakst voorkomt met '14474' waarnemingen. Laten we dus als baseline model pakken 'Alle oplostijden kosten  '3.0' minuten', laten we nu de score bekijken.

In [8]:
#
def range_score(van, tot):
    # bepaal percentage repareertijd o.b.v. gewenenste randvoorwaarden.
    range_values = df[(df['hersteltijd'] >= van) & (df['hersteltijd'] < tot)]
    score = len(range_values) / len(df) * 100
    print(round(score, 4), '% van de storingen zijn binnen {} tot {} minuten opgelost.'.format(van, tot))

In [9]:
range_score(2.9, 3.1)
range_score(0, 5)
range_score(float(input('geef starttijd: ')), float(input('geef eindtijd: ')))

5.5854 % van de storingen zijn binnen 2.9 tot 3.1 minuten opgelost.
14.7256 % van de storingen zijn binnen 0 tot 5 minuten opgelost.
geef starttijd: 0
geef eindtijd: 30
49.0113 % van de storingen zijn binnen 0.0 tot 30.0 minuten opgelost.


In [10]:
# x en y vectors
x = df[['hersteltijd']]
y = df[['hersteltijd']].apply(lambda yy: yy * 0 + 5.0)

print(np.sqrt(mean_squared_error(x, y)))

69.19659563125424
